### PR TITLE
Draft: Coq plugin

### PR DESCRIPTION
**Pull request by thibautbenjamin**
A Coq plug-in to export CaTT terms into idenitty proofs in Coq

### CODE REVIEW

**Comment by thibautbenjamin** [comment]
@regular-citizen This is the branch for the coq plugin based on catt. Can you have a look at it and tell me what you think of it? In particular, I would like to know if you have any suggestions when it comes to usability and practical use

**Comment by regular-citizen** [comment]
To be added: implicit CaTT arguments should be implicit in CoQ export.

File: coq_plugin/src/export.ml [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+  val tm : Environ.env -> evar_map -> ctx -> tm -> evar_map * econstr
+  val coh : Environ.env -> evar_map -> Coh.t -> evar_map * econstr
+end = struct
+
+  let catt_to_coq_db ctx var =
+    match var with
+    | Var.Db n -> let l = (List.length ctx - n) in EConstr.mkRel l, l
+    | _ -> assert false
+
+  let induction_vars ps =
```
Comment (regular-citizen) [inline coq_plugin/src/export.ml:46]: This is thoroughly unscrutable

File: coq_plugin/src/export.ml [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+      match fst (List.assoc var ctx) with
+      | Obj -> assert false
+      | Meta_ty _ -> Error.fatal "unresolved meta variable"
+      | Arr(ty,s,_) ->
+        match s with
+        | Var s -> (var,s,ty)
+        | _ -> assert false
+    in List.map find_data list_vars
+
+  (* let rec print_list l = *)
```
Comment (regular-citizen) [inline coq_plugin/src/export.ml:75]: Dead code, can delete

File: coq_plugin/src/export.ml [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+  let register_tm tm_name =
+    let env = Global.env () in
+    let sigma = Evd.from_env env in
+    let sigma,body =
+      match Environment.val_var (Var.Name tm_name) with
+      | Coh c -> Translate.coh env sigma c
+      | Tm (ctx,tm) -> Translate.tm env sigma ctx tm
+    in
+    (* let _ = Feedback.msg_debug *)
+    (*   (str "body:" ++ *)
```
Comment (regular-citizen) [inline coq_plugin/src/export.ml:202]: Dead code

**Review by regular-citizen — APPROVED** [review]
Approved with minor changes, most of the code is pretty unscrutable and would benefit from comments

**Comment by thibautbenjamin** [comment]
I cleaned up the code, added comments and opened a new issue with suggested improvements for this plugin.

Merging this now

**Pull request merged**